composables/useCopyName.ts

```diff
@@ -0,0 +1,46 @@
+import { FileObject } from "@supabase/storage-js";
+
+export const useCopyName = () => {
+  const copyName = (fileName: string, files: FileObject[]) => {
+    const fileETag = () =>
+      files.filter((file) => file.name === fileName)[0].metadata.eTag;
+
+    // sorted files with the same etag, where file at 0 index is original file and at 1 index is the latest copy (highest number)
+    const filesWithSameETag = () =>
+      files
+        .filter((file) => file.metadata.eTag === fileETag())
+        .sort((a, b) => b.name.localeCompare(a.name));
+
+    const originalFile = filesWithSameETag()[0].name;
+
+    const name = useFileName(fileName);
+
+    let newCopyName: string | null;
+
+    if (
+      (filesWithSameETag().length === 1 && fileName === originalFile) ||
+      (filesWithSameETag().length > 1 && fileName !== originalFile)
+    ) {
+      newCopyName = `${name.name}_(1)${name.extension}`;
+    } else if (filesWithSameETag().length > 1 && fileName === originalFile) {
+      const prevCopy = filesWithSameETag()[1].name || null;
+
+      const prevCopyFullName = useFileName(prevCopy || "");
+
+      const prevCopyNumber = Number(
+        prevCopyFullName.name.substring(
+          prevCopyFullName.name.length - 2,
+          prevCopyFullName.name.length - 1
+        )
+      );
+
+      newCopyName = `${name.name}_(${prevCopyNumber + 1})${name.extension}`;
+    } else {
+      newCopyName = null;
+    }
+
+    return newCopyName;
+  };
+
+  return { copyName };
+};
```

Comment:
Here is the logic for names of created copies. Each file has a eTag key which is the same to every created copy of the same file, so I use to it check if the file already has some copies created. Generally first copy get name: name_(1).extension, if it already exists - then name_(2).extension, etc. But when I copy name_(2).extension for the fist time - I will get name_(2)_(1).extension.

Reply:
looks good 👌 imo better than making recursive calls 💪